### DISCUSSION TITLE
Elastic-Fleet Error and Agent Fail

### DISCUSSION THREAD

**Discussion by rmchau** [Security-Onion-Solutions/securityonion#11787] [2.4]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

16

### RAM

134GB

### Storage for /

100gb

### Storage for /nsm

180

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hi,

I am having issue with the Elastic-fleet service. I tried reinstalling SO to resolve the problem but it persist and I am not sure what has been missed.

When install the agent on a machine I get the below Error:

Context="Installing in non-interactive mode." ErrorDetails="exit status 1"

Running sudo salt-call state.highstate , There are 2 Errors:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/cffcd506-f972-480c-8727-bd6fdbcb17b9)
 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/3530932f-0b97-43cd-9deb-7ddb63702fd8)

The only Errors I can see in the logs are below. I am not sure if certificate is an issue that can causing this?
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/69c9aa99-5ef4-4ca3-8514-9fc961aaa691)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/948829dc-fdfa-435a-bcf8-6c256d334262)

But the Grid shows all services related to ElasticSearch are running as they should:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/14ebcbc2-093a-48f1-a56a-d6b82c42145b)

I also get the below Error when trying to restart the service. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/69132727-b0a9-496c-9705-d1da5348fc4c)

On the Agents I get these below errors: (out-of-date Policy and Unhealthy)
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/f216f04f-75cd-4539-a6ae-77ce2414e07c)

and the elastic-defend-endpoints is unable to connect to ElasticSearch.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/47691877/c7e83276-c623-4e6b-a1f5-89fb38219ef6)

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by robbiemarshall** [comment]
Did you add the IP/subnet of your endpoints in the firewall per the documentation?  https://docs.securityonion.net/en/2.4/elastic-agent.html#deployment
